app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/survey/TerminationSurveyViewModel.kt

```diff
@@ -136,20 +140,33 @@ internal class TerminationSurveyPresenter(
             }
             currentState = currentState.copy(
               actionButtonLoading = false,
-              errorWhileLoadingNextStep = GENERAL,
+              errorWhileLoadingNextStep = true,
             )
             loadBetterQuotesSource = null
           },
           ifRight = { changeTierIntent ->
             if (changeTierIntent.quotes.isEmpty()) {
+              val optionsToDisable = currentReasonsWithFeedback.filter {
+                it.key.suggestion is SurveyOptionSuggestion.Action.DowngradePriceByChangingTier ||
+                  it.key.suggestion is SurveyOptionSuggestion.Action.UpgradeCoverageByChangingTier
+              }.keys
+              optionsToDisable.forEach {
+                if (currentReasonsWithFeedback.contains(it)) {
+                  currentReasonsWithFeedback.remove(it)
+                  val newOption = it.copy(isDisabled = true)
+                  currentReasonsWithFeedback[newOption] = null
+                }
+              }
```

Comment:
```suggestion
              Snapshot.withMutableSnapshot {
                  val optionsToDisable = currentReasonsWithFeedback.filter {
                    it.key.suggestion is SurveyOptionSuggestion.Action.DowngradePriceByChangingTier ||
                      it.key.suggestion is SurveyOptionSuggestion.Action.UpgradeCoverageByChangingTier
                  }.keys
                  optionsToDisable.forEach {
                    if (currentReasonsWithFeedback.contains(it)) {
                      currentReasonsWithFeedback.remove(it)
                      val newOption = it.copy(isDisabled = true)
                      currentReasonsWithFeedback[newOption] = null
                    }
                  }
              }
```
Probably a good idea to make sure that since we are doing many mutations here inside this one mutable set, we do it all and submit it in one go through the use of `Snapshot.withMutableSnapshot` instead of the state being mutated in various ways in sequence, meaning that while we're in the middle of this process we do not want to accidentally have it show half-done to the UI

Not sure how likely this is to happen and be a problem, but I still would suggest we do this anyway.